### PR TITLE
feat!: redirect folders to `/opt/containerbase`

### LINKED ISSUE #2078: Support readonly root filesystem
### general

- `/usr/local/bin` is symlinked to `/opt/containerbase/bin`
- `/usr/local/lib` is symlinked to `/opt/containerbase/lib`
- tool prepare state is stored at `/var/lib/containerbase`
- all cache files are stored at `/tmp/containerbase`
- files and folders in user home are symlinked to `/tmp/containerbase/cache` (eg `.cache`, `.npmrc`, `.npm`)

### `binarySource=global`
- needs `/tmp` writable

### `binarySource=install`
- needs `/tmp` writable
- needs `/opt/containerbase` writable
---
*related issues*
- #101
- #380
- #1384
- #1385

### PR DESCRIPTION
**Pull request by viceice**
It should now be possible to use a readonly filesystem as root.
You need to mount writable volumes to `/tmp` and `/opt/containerbase`.

---
- #2078

⚠️ Additional commit message:
```
BREAKING CHANGE: Containerbase binaries are moved, read custom image guide.
The paths `/usr/local/{bin,lib}` are now symlinks to `/opt/comntainerbase/{bin,lib}`.
```

### CODE REVIEW

**Comment by rarkins** [comment]
Can it be tested in practice or are we limited to theory?

**Comment by viceice** [comment]
> Can it be tested in practice or are we limited to theory?

Will prepare some test which can be run manually.

File: test/node/Dockerfile [hunk 1]
```diff
@@ -16,8 +16,12 @@ ARG TARGETARCH
 COPY dist/docker/ /
 COPY dist/cli/containerbase-cli-${TARGETARCH} /usr/local/containerbase/bin/containerbase-cli
 
+RUN set -ex; ls -la /usr/local/bin/; ls -la /usr/local/sbin;
+
```
Comment (viceice) [inline test/node/Dockerfile:20]:
```suggestion
```

File: test/node/Dockerfile [hunk 2]
```diff
@@ -16,8 +16,12 @@ ARG TARGETARCH
...
 RUN install-containerbase
 
+RUN set -ex; ls -la /usr/local/bin/; ls -la /usr/local/sbin; ls -la /opt/containerbase/bin;
+
```
Comment (viceice) [inline test/node/Dockerfile:24]:
```suggestion
```

**Pull request merged**